test/integration/isolated/udproute_test.go

```diff
@@ -0,0 +1,637 @@
+//go:build integration_tests
+
+package isolated
+
+import (
+	"context"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	ktfkong "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong"
+	"github.com/kong/kubernetes-testing-framework/pkg/utils/kubernetes/generators"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	gatewayclient "sigs.k8s.io/gateway-api/pkg/client/clientset/versioned"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/gatewayapi"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/util/builder"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/integration/consts"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/helpers"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/testlabels"
+)
+
+const (
+	// coreDNSImage is the image and version of CoreDNS that will be used for UDP testing.
+	coreDNSImage = "registry.k8s.io/coredns/coredns:v1.8.6"
+)
+
+func TestUDPRoute(t *testing.T) {
+	const (
+		corefile = `
+	.:53 {
+		errors
+		health
+		ready
+		kubernetes cluster.local in-addr.arpa ip6.arpa {
+		   pods insecure
+		   fallthrough in-addr.arpa ip6.arpa
+		   ttl 5
+		}
+		forward . /etc/resolv.conf {
+		   max_concurrent 1000
+		}
+		cache 1
+		loop
+		reload
+		loadbalance
+		hosts {
+		  10.0.0.1 konghq.com
+		  fallthrough
+		}
+	}
+	.:9999 {
+		errors
+		health
+		ready
+		kubernetes cluster.local in-addr.arpa ip6.arpa {
+		   pods insecure
+		   fallthrough in-addr.arpa ip6.arpa
+		   ttl 5
+		}
+		forward . /etc/resolv.conf {
+		   max_concurrent 1000
+		}
+		cache 1
+		loop
+		reload
+		loadbalance
+		hosts {
+		  10.0.0.1 konghq.com
+		  fallthrough
+		}
+	}
+	`
+		testdomain = "konghq.com"
+	)
+
+	t.Parallel()
+
+	var udprouteParentRefs []gatewayapi.ParentReference
+
+	fEssentials := features.
+		New("essentials").
+		WithLabel(testlabels.NetworkingFamily, testlabels.NetworkingFamilyGatewayAPI).
+		WithLabel(testlabels.Kind, testlabels.KindUDPRoute).
+		Setup(SkipIfRouterNotExpressions).
+		WithSetup("deploy kong addon into cluster", featureSetup()).
+		WithSetup("inject Klient into test context", injectKlient()).
```

Comment:
I couldn't find any place where we'd use the injected controller-runtime client. As I understand, we can get one by calling `cfg.Client().Resources()` and that's what we do in some places. Isn't `injectKlient` redundant?

Reply:
Those are slightly different, I believe my original intention was to take into account e.g. in cluster config (for whatever reason) which `conf.New()` can do: https://github.com/kubernetes-sigs/e2e-framework/blob/262cac32d35e71d9f9591cf88c2126e2afe0a9f0/klient/conf/config.go#L36-L63
(used in https://github.com/Kong/kubernetes-ingress-controller/blob/134eb00a9876fef433213abc394a159d06ead61e/test/integration/isolated/suite_test.go#L338) but we pass it the kubeconfig explicitly anyway 🤔 

Removing this as 1c2eb68f8